### PR TITLE
cclient: implemented iota_api_attach_to_tangle

### LINKED ISSUE #42: Implement attach to tangle

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR fixed #42 

# Test Plan:
CI must pass

### CODE REVIEW

File: cclient/request/attach_to_tangle.h [hunk 1]
```diff
@@ -14,6 +14,8 @@ extern "C" {
 
 #include "types/types.h"
 
+#define ATTACH_TO_TANGLE_MIN_MWM 18
```
Comment (tsvisabo) [inline cclient/request/attach_to_tangle.h]: mainnet is 14

File: cclient/request/attach_to_tangle.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (*req) {
+    char_buffer_free((*req)->trunk);
+    char_buffer_free((*req)->branch);
+    utarray_free((*req)->trytes);
+    free(*req);
+    *req = NULL;
+  }
+}
+
+retcode_t attach_to_tangle_req_set_trunk(attach_to_tangle_req_t* req,
```
Comment (tsvisabo) [inline cclient/request/attach_to_tangle.c:32]: could you use address_t and memcpy?

File: cclient/request/attach_to_tangle.h [hunk 3]
```diff
@@ -26,14 +28,25 @@ typedef struct {
...
 
 } attach_to_tangle_req_t;
 
+attach_to_tangle_req_t* attach_to_tangle_req_new();
+void attach_to_tangle_req_free(attach_to_tangle_req_t** req);
+retcode_t attach_to_tangle_req_set_trunk(attach_to_tangle_req_t* req,
+                                         const char* trunk);
+retcode_t attach_to_tangle_req_set_branch(attach_to_tangle_req_t* req,
+                                          const char* branch);
+void attach_to_tangle_req_set_mvm(attach_to_tangle_req_t* req,
```
Comment (tsvisabo) [inline cclient/request/attach_to_tangle.h]: typo: mwm

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/request/attach_to_tangle.h (hunk 1 above)
Comment (oopsmonk) [reply cclient/request/attach_to_tangle.h]: ah, 18 is in the past.
Docs need an update though. 
https://iota.readme.io/v1.5.0/reference#attachtotangle

Comment (tsvisabo) [reply cclient/request/attach_to_tangle.h]: yeah :)

File: cclient/request/attach_to_tangle.c (hunk 2 above)
Comment (oopsmonk) [reply cclient/request/attach_to_tangle.c:32]: try to keep plain text in the JSON serialization & test cases, so we can do the flex trits conversion as needed. I'm not sure about keeping the flex_tirits in a structure is better, but I will figure it out later.

Comment (tsvisabo) [reply cclient/request/attach_to_tangle.c:32]: can you do that and using `cJSON_CreateArray` instead of `cJSON_CreateString` ?

**Review by tsvisabo — APPROVED** [review]

File: cclient/response/attach_to_tangle.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const char* attach_to_tangle_res_trytes_at(attach_to_tangle_res_t* res,
+                                           int index) {
+  if (utarray_len(res) > index) {
+    char** p = NULL;
+    p = (char**)utarray_eltptr(res, index);
+    if (p) {
+      return *p;
+    }
+  }
+  return '\0';
```
Comment (thibault-martinez) [inline cclient/response/attach_to_tangle.c]: Shouldn't this return `NULL` ? It's expected to return a `char *`, you're returning a `char`.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/response/attach_to_tangle.c (hunk 4 above)
Comment (oopsmonk) [reply cclient/response/attach_to_tangle.c]: ah, silly mistake. thanks.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**